Dynatrace/dynatrace-operator
# Remove db from csi-server

## Description

[DAQ-1472](https://dt-rnd.atlassian.net/browse/DAQ-1472)

No more database usage in the `server` container.
- It relies solely on the filesystem
   - Creation time (`ModTime`) of the app-volume dir to timeout, instead of counting attempts
   - Well-known paths (latest binary path, managed by the `provisioner`, `appmounts` folder for all the directories for the mounted volumes) instead of a database

## How can this be tested?

Both host and app mounts should work as usual, if you check the fs of the `server`, you will see things are in a different place

⚠️ The migration logic for older/already mounted volumes is not implemented, DON'T upgrade/replace your csi-driver before unmounting everything (didn't try this, so I don't know what would happen)

[DAQ-1472]: https://dt-rnd.atlassian.net/browse/DAQ-1472?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
